flow/project.py

```diff
@@ -826,9 +826,16 @@ def _map_scheduler_jobs(scheduler_jobs):
                 expanded = JobOperation.expand_id(name)
                 yield expanded['job_id'], expanded['operation-name'], sjob
 
-    def _get_operations_status(self, job):
+    def _get_operations_status(self, job, cached_status=None):
         "Return a dict with information about job-operations for this job."
-        cached_status = self.document.get('_status', dict())
+        if cached_status is None:
```

Comment:
To reduce complexity please convert `cached_status` to a required argument for this internal function.